src/components/BannerConductor/BannerConductor.vue

```diff
@@ -77,6 +79,8 @@ async function closeHandler( closeEvent: TrackingEvent<void> ): Promise<any> {
 	await stateMachine.changeState( stateFactory.newClosedState( closeEvent ) );
 }
 
+onUnmounted( timer.clearAll );
```

Comment:
The Banner Conductor is never unmounted, currently the whole code stays in the DOM of the page, just hidden. All the timers continue to tick, leaking memory.

I can see three ways moving forward:
1. Make all components responsible to stop the timers they started (like `KeenSlider`). this can be confusing and hard, esp. if they are not unmounted but just hidden but the fact that they no longer displayed needs to reach them. Worst option IMHO.
2. Pass timer as a dependency to the "final" states of the state machine (closed & not shown) and call `timer.clearAll` there.
3. Add code to the "final" states of the state machines that allows to call [`app.unmount()`](https://vuejs.org/api/application.html#app-unmount) to thoroughly remove the banner and clear up its resources. Since [getting the app instance from a component is being removed from Vue](https://stackoverflow.com/q/75446100/130121), the best way to do this is probably via another injection, e.g. `app.provide( 'unmountBanner', () => app.unmount() )`

Reply:
The timers should be cleared in the states anyway, as that's where the business logic for the conductor should be. I don't know why I put it into the components.

We should chat about cleaning up the banner properly on close though, as it would free up some browser memory